packages/dds/tree/src/test/simple-tree/schemaFactory.spec.ts

```diff
@@ -143,6 +143,30 @@ describe("schemaFactory", () => {
 		assert.equal(foo, "foo");
 	});
 
+	it("Optional fields", () => {
+		const factory = new SchemaFactory("test");
+		const foo = factory.object("foo", {
+			x: factory.optional(factory.number),
+		});
+
+		const _check1 = new foo({});
+		const _check2 = new foo({ x: undefined });
+		const _check3 = new foo({ x: 1 });
```

Comment:
If you are just going to test one syntax (which is fine IMO), I think it would be better to test the more compatible one (which supports customizations, and doesn't break when package exported if recursive, and gets an explicit name in IntelliSense). Also capitalizing class names is conventional:

```suggestion
		class Foo extends factory.object("foo", {
			x: factory.optional(factory.number),
		}) {}

		const _check1 = new Foo({});
		const _check2 = new Foo({ x: undefined });
		const _check3 = new Foo({ x: 1 });
```

Our schema are a bit fussy with format and we have had some issues with people copying non-conventional examples for use in tests, so I'd like to try and keep all usages in our preferred/recommended format unless the test has a specific reason to test the other one (ex: documented to be testing POJO mode)

Reply:
Same applies to the test below.